satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,617 @@
+don't have to log in more than once when they try to access to a different satellite that the one
+that they logged in for satisfying the _users stories_.
+
+For Storj DCS, we determined that the we want to authenticate the users through Oauth2 with their
+Storj email account, which is hosted by Google Workspace. We are going to authorize assigning them
+to different groups and mapping those groups to the back office roles.
+
+For authenticating the users we will use one [Oauth2-proxy](https://oauth2-proxy.github.io/oauth2-proxy/)
+for each satellite and for the "service" that will show the list of satellites, henceforth "selector
+service".
+
+All the Oauth2-proxies must use the same Oauth2 client credentials for not requiring the users to
+log in when they access to a different satellite or from the "selector service" once they logged in
+successfully in one of them and all the satellites and the "selector service" must be hosted under
+the same base domain or sub-domain. You can see [this proof of concept](https://github.com/ifraixedes/poc-oauth2-proxies-single-auth)
+for how to achieve it.
+
+##### API implementation
+
+We are going to build the API from the scratch because we want to use
+[API generator](https://github.com/storj/storj/tree/main/private/apigen).
+
+We may need to improve the API generator and add more functionalities to support every feature that
+we need to implement the back office API.
+
+The API has to perform the operations based on the usr roles specified in the
+_Authentication / Authorization_ section and be able to configure the groups associated to them.
+
+The API has to use the logging and metrics systems that we use for the rest of the satellite and
+using the similarly.
+
+Because the operations that the users can perform with the back office, the API MUST log all the
+operations performed by the users using a specific "log name" to ease filtering only them. We are
+using the
+[same mechanism for logging all the requests to the satellite console](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/server.go#L1170).
+Every log must include the API call and the operator's email who has executed it, but it doesn't
+have to contain any data from the request related to the customer, including URL paths that are
+parametrized with users'd data (e.g. email address) for not leaking any personal data into the logs.
+
+We'll delete the current admin API once this new one is rolled on.
+
+##### Web UI
+
+We have to build a new Web UI from scratch with the Vue framework and the Vuetify component
+framework. The Web UI is already in development and it almost have every component and page that the
+back office requires, you can see it on https://storj.github.io/admin-ui/login.
+
+We need to add the API calls and UI logic (client side validations, operation confirmation, etc.)
+and all the Javascript client side code that those require.
+
+We use the Javascript API client generated by
```

Comment:
here are some additional links of actual implementation of the generator which could be helpful. Of course, we will want a separate implementation for the back office, but these could be a good reference:
* [API definition](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/consoleapi/gen/main.go). In theory, this is the only one you need to modify manually. Everything else is generated at the paths defined in this doc.
* [Generated backend code](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/consoleapi/api.gen.go) based on definition
* [Generated documentation](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/consoleapi/apidocs.gen.md) based on definition
* [Generated typescript/frontend code](https://github.com/storj/storj/blob/main/web/satellite/src/api/v0.gen.ts) based on definition.

Reply:
I add those references to the document.